crates/sui-framework/packages/sui-framework/sources/alias.move

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+#[allow(unused_use)]
+// Module for managing address alias configurations.
+module sui::alias;
+
+use sui::derived_object;
+use sui::party;
+use sui::vec_set;
+
+#[error(code = 0)]
+const ENotSystemAddress: vector<u8> = b"Only the system can create the alias state object.";
+#[error(code = 1)]
+const ENoSuchAlias: vector<u8> = b"Given alias does not exist.";
+#[error(code = 2)]
+const EAliasAlreadyExists: vector<u8> = b"Alias already exists.";
+#[error(code = 3)]
+const ECannotRemoveLastAlias: vector<u8> = b"Cannot remove the last alias.";
+#[error(code = 4)]
+const ETooManyAliases: vector<u8> = b"The number of aliases exceeds the maximum allowed.";
+
+const CURRENT_VERSION: u64 = 0;
+
+const MAX_ALIASES: u64 = 8;
+
+/// Singleton shared object which manages creation of AddressAliases state.
+/// The actual alias configs are created as derived objects with this object
+/// as the parent.
+public struct AliasState has key {
+    id: UID,
+    // versioned to allow for future changes
+    version: u64,
+}
+
+#[allow(unused_function)]
+/// Create and share the AliasState object. This function is called exactly once, when
+/// the alias state object is first created.
+/// Can only be called by genesis or change_epoch transactions.
+fun create(ctx: &TxContext) {
+    assert!(ctx.sender() == @0x0, ENotSystemAddress);
+
+    let self = AliasState {
+        id: object::alias_state(),
+        version: CURRENT_VERSION,
+    };
+    transfer::share_object(self);
+}
+
+/// Tracks the set of addresses allowed to act as a given sender.
+///
+/// An alias allows transactions signed by the alias address to act as the
+/// original address. For example, if address X sets an alias of address Y, then
+/// then a transaction signed by Y can set its sender address to X.
+public struct AddressAliases has key {
+    id: UID,
+    aliases: vec_set::VecSet<address>,
+}
+
+/// Internal key used for derivation of AddressAliases object addresses.
+public struct AliasKey(address) has copy, drop, store;
+
+/// Provides the initial set of address aliases for the sender address.
+///
+/// By default, an address is its own alias. However, the original address can
+/// be removed from the set of allowed aliases after initialization.
+entry fun init_aliases(alias_state: &mut AliasState, ctx: &TxContext) {
+    assert!(!derived_object::exists(&alias_state.id, ctx.sender()), EAliasAlreadyExists);
```

Comment:
shouldn't this have the same key as the claim call? also why is this necessary, wouldn't claim just abort anyway if the id is already claimed?

Reply:
thanks yes, fixed - this was just to provide a better error message